ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -1,307 +1,97 @@
+data ValidateBlocks = ValidateAll | MinimumValidation
 
-processAll :: forall blk. (HasHeader blk, ImmDbSerialiseConstraints blk)
-           => ImmDB IO blk
-           -> ResourceRegistry IO
-           -> (blk -> IO ())
-           -> IO ()
-processAll immDB rr callback = do
-    tipPoint <- getPointAtTip immDB
-    case pointToWithOriginRealPoint tipPoint of
-      Origin -> return ()
-      At tip -> do
-        Right itr <- stream immDB rr GetBlock
-          (StreamFromExclusive genesisPoint)
-          (StreamToInclusive tip)
-        go itr
-  where
-    go :: Iterator (HeaderHash blk) IO (IO blk) -> IO ()
-    go itr = do
-        itrResult <- ImmDB.iteratorNext itr
-        case itrResult of
-          IteratorExhausted   -> return ()
-          IteratorResult mblk -> mblk >>= \blk -> callback blk >> go itr
+data BlockType =
+    ByronBlock   ByronBlockArgs
+  | ShelleyBlock ShelleyBlockArgs
+  | CardanoBlock CardanoBlockArgs
 
 {-------------------------------------------------------------------------------
-  Command line args
+  Parsing
 -------------------------------------------------------------------------------}
 
-data CmdLine = CmdLine {
-      clConfig    :: [FilePath]
-    , clIsMainNet :: Bool
-    , clImmDB     :: FilePath
-    , clBlockType :: Maybe BlockType
-    , clAnalysis  :: AnalysisName
-    } deriving Show
-
-data BlockType = Byron | Shelley | Cardano
-  deriving Show
-
 parseCmdLine :: Parser CmdLine
 parseCmdLine = CmdLine
-    <$> many (strOption (mconcat [
-            long "config"
-          , help "Path to config file or files. Multiple occurences of the\
-                 \option can be used"
-          , metavar "PATH"
-          ]))
-    <*> flag True False (mconcat [
-            long "testnet"
-          , help "The DB contains blocks from testnet rather than mainnet"
-          ])
-    <*> strOption (mconcat [
+    <$> strOption (mconcat [
             long "db"
-          , help "Path to the chain DB (parent of \"immutable\" directory)"
+          , help "Path to the chain DB"
           , metavar "PATH"
           ])
-    <*> parseBlockType
+    <*> switch (mconcat [
+            long "verbose"
+          , help "Enable verbose logging"
+          ])
+    <*> switch (mconcat [
+            long "onlyImmDB"
+          , help "Validate only the immutable DB (e.g. do not do ledger validation)"
+          ])
+    <*> parseValidationPolicy
+    <*> blockTypeParser
     <*> parseAnalysis
 
-parseBlockType :: Parser (Maybe BlockType)
-parseBlockType = asum [
-      flag' (Just Byron) $ mconcat [
-          long "byron"
-        , help "A Byron network"
-        ]
-    , flag' (Just Shelley) $ mconcat [
-          long "shelley"
-        , help "A Shelley network"
+parseValidationPolicy :: Parser (Maybe ValidateBlocks)
+parseValidationPolicy = parseMaybe $ asum [
+      flag' ValidateAll $ mconcat [
+          long "validate-all"
```

Comment:
```suggestion
          long "validate-all-blocks"
```
Otherwise it might be confusing to users that want to use this tool to do ledger validation.